Add support for NetBSD/aarch64-be (big-endian arm64).

he32 opened the pull request.

he32 committed:
Add support for NetBSD/aarch64-be (big-endian arm64).

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4908730077/jobs/8764681288) [(plain)](https://github.com/rust-lang/rust/commit/6ef377cc41dbc7ba9455d0b44f13c3c1a024b0cf/checks/13296755015/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
platform support check
   Compiling tier-check v0.1.0 (/checkout/src/tools/tier-check)
    Finished release [optimized] target(s) in 0.85s
     Running `obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/tier-check /checkout/src/doc/rustc/src/platform-support.md /checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/rustc`
error: target `aarch64_be-unknown-netbsd` is missing from platform-support.md
If this is a new target, please add it to /checkout/src/doc/rustc/src/platform-support.md.
```

</details>

b-naber commented:
r? rust-lang/compiler

he32 commented:
> These commits modify **compiler targets**.
> (See the [Target Tier Policy]
> (https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html).)

Hm...  Yes...  I see there is some documentation I need to fill
out, will take a stab drafting that and including it in an
amended pull request.

Regards,

- Håvard

WaffleLapkin commented:
Marked this as waiting on author, feel free to `@rustbot ready` once you filled out everything regarding the target tier policy and fixed CI.
r? compiler

Urgau commented:
As said by CI you are missing an the entry in the [platform support page](https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support.md) (`src/doc/rustc/src/platform-support.md`), with the page for the target in `src/doc/rustc/src/platform-support/`. Take a look at the [template](https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support/TEMPLATE.md) and an [example](https://github.com/rust-lang/rust/pull/108792/files#diff-98c317f447306855c6e723385f2a38b516acb20296d4131b5572a2ca1b112add) for what is roughly expected.

he32 committed:
platform-support.md: document the various NetBSD targets.

This is slightly patterned after what OpenBSD has done.

This is a step along the path to reduce the number and amount
of diffs that pkgsrc carries around for rust, and this documents
also some parts which have not yet been upstreamed (mipsel-*).

he32 commented:
I committed above a suggested update to `platform-support.md`, and added a `netbsd.md` file, partially patterned after the OpenBSD file.  Almost all of the NetBSD targets are suggested as "Tier 3" systems, and do not have CI setups configured.
Hope this is a step in the right direction.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5094649139/jobs/9158720768) [(plain)](https://github.com/rust-lang/rust/commit/0a1e42e0e27c46b245fbd025af353b9031217fae/checks/13797364768/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:8e5e7e5ab8b370d6c329ec480221332ada57f0ab)
Download action repository 'rust-lang/simpleinfra@master' (SHA:cdb59432ef9e018dd6db5879e720eccc3289da68)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: mingw-check-tidy
---
Building wheels for collected packages: reuse
  Building wheel for reuse (pyproject.toml): started
  Building wheel for reuse (pyproject.toml): finished with status 'done'
  Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=180115 sha256=c3e1cf75e5aea7348ad201d16633573bc9f3aa58c8e29a6ca459f306300d6c9d
  Stored in directory: /tmp/pip-ephem-wheel-cache-tqh3r9wq/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
  Attempting uninstall: setuptools
    Found existing installation: setuptools 59.6.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
Successfully tagged rust-ci:latest
Built container sha256:f14905d548c4fed093f32df63ae48cdb4de5b589590f66027415b07351e4964a
Uploading finished image to https://ci-caches.rust-lang.org/docker/e51156f19850ce886cec818e46dc2a021e0aa7c270d15673e8fe74cd8522fc8ac3995109aebb688ee49ed586735ed4cf5f8c06d44c48298fa09c35ae4b082281

<botocore.awsrequest.AWSRequest object at 0x7f38a6aaac90>
gzip: stdout: Broken pipe
xargs: docker: terminated by signal 13
[CI_JOB_NAME=mingw-check-tidy]
[CI_JOB_NAME=mingw-check-tidy]
---
##[endgroup]
fmt check
tidy check
tidy: Skipping binary file check, read-only filesystem
tidy error: /checkout/src/doc/rustc/src/platform-support/netbsd.md: too many trailing newlines (2)
some tidy checks failed
```

</details>

he32 commented:
@rustbot ready (?)

wesleywiser commented on src/doc/rustc/src/platform-support/netbsd.md:
```diff
@@ -0,0 +1,108 @@
+The pkgsrc rust package has a few files to assist with building
+pkgsrc packages written in rust, ref. the `rust.mk` and `cargo.mk`
+Makefile fragments in the `lang/rust` package.
+
```
```suggestion
```

wesleywiser committed:
Remove extra trailing newline

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5346903376/jobs/9694581217) [(plain)](https://github.com/rust-lang/rust/commit/da05ed36836077392384573d5277f8a01bbe17db/checks/14474512455/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Rustbook (x86_64-unknown-linux-gnu) - style-guide
##[group]Building stage0 tool linkchecker (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.09s
##[endgroup]
rustc/platform-support.html:349: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:356: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:364: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:372: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:385: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:389: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:402: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:411: broken link - `rustc/platform-support/netbsd.html`
rustc/platform-support.html:435: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8425: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8432: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8440: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8448: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8461: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8465: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8478: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8487: broken link - `rustc/platform-support/netbsd.html`
rustc/print.html:8511: broken link - `rustc/platform-support/netbsd.html`
number of HTML files scanned: 33980
number of HTML redirects found: 10327
number of links checked: 2974957
number of links ignored due to external: 57394
```

</details>

he32 commented:
> The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5346903376/jobs/9694581217) [(plain)](https://github.com/rust-lang/rust/commit/da05ed36836077392384573d5277f8a01bbe17db/checks/14474512455/logs)
> Click to see the possible cause of the failure (guessed by this bot)

I'll admit that I don't quite understand the criterion for the "broken link" messages.
The links I inserted in the source document are all possible to visit, so what is "broken"
about them?

oli-obk commented:
I don't know if the rustc guide has any guidance for that, but I'd just look at how other links work and check if

a) the link should be relative to the file
b) the link should have a different extension

Urgau commented:
@he32 I think you are missing the definition in the `SUMMARY.md` file: [`src/doc/rustc/src/SUMMARY.md`](https://github.com/rust-lang/rust/blob/da05ed36836077392384573d5277f8a01bbe17db/src/doc/rustc/src/SUMMARY.md); without it mdbook [does not know](https://rust-lang.github.io/mdBook/format/summary.html) how to include files (or anything for that matter).

he32 committed:
SUMMARY.md: add entry for NetBSD.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5370714249/jobs/9742994417) [(plain)](https://github.com/rust-lang/rust/commit/7a07ffdbd880f0c983ae774f64138489ce88faf0/checks/14535839343/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
platform support check
   Compiling tier-check v0.1.0 (/checkout/src/tools/tier-check)
    Finished release [optimized] target(s) in 0.61s
     Running `obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/tier-check /checkout/src/doc/rustc/src/platform-support.md /checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/rustc`
error: target `mipsel-unknown-netbsd` is in platform-support.md, but does not appear in the rustc target list
If the target has been removed, please edit /checkout/src/doc/rustc/src/platform-support.md and remove the target.
error: target `i586-unknown-netbsd` is in platform-support.md, but does not appear in the rustc target list
If the target has been removed, please edit /checkout/src/doc/rustc/src/platform-support.md and remove the target.
```

</details>

he32 committed:
platform-support.md: remove references to not-yet upstreamed targets.

bors commented:
:pushpin: Commit abb43958a97f1619085b0ec4697d00b86bdb8275 has been approved by `oli-obk`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=oli-obk abb43958a97f1619085b0ec4697d00b86bdb8275 -->
<!-- homu: {"type":"Approved","sha":"abb43958a97f1619085b0ec4697d00b86bdb8275","approver":"oli-obk","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
